### PR TITLE
[P2 tests] [Team 10] Add Task 2 Merge Segments Tests

### PR DESCRIPTION
**Pull request by KaifuJ**
What's your team number?
Team 10

What is the functionality being tested?
Merge segements on disk, including dictionaries and posting lists.

Describe your tests briefly:
test1 tests if the implementation can merge two segments (containing dictionaries and posting lists) properly. The documents is a little complex to test if it can be indexed correctly.
test2 tests if multiple segments can be merged pair-wise, and if the doc id can be modified correspondingly.
Both of the tests test if addDocument(), flush(), and mergeAllSegments() functions are implemented correctly.

Read / write counter check:
Read and write counters represent the times we have read and writen file pages, which are in the PageFileChannel class. The value of read counter should be about the number of pages of all the input segments. The value of write counter should be about the number of pages of all the output segments. To get the number of pages, we need the getNumPages() function in PageFileChannel. However, getNumPages() is not a static function, which means we cannot access it in Test function, since we don't have PageFIleChannel object in Test function. So I think we cannot test the value of read / write counters.

Does each test case have comments/documentation?
Yes. Actually the code is quite self explained.

### CODE REVIEW

**Comment by omidkv** [comment]
Response from team 14:

Your test looks alright. For test 1, I think one issue that we saw was that if the document numbering system is different for everyone's personal representation this test may cause issues for them. It makes sense that everyone increases document numbers as they go along but it could be a problem. I think that we also need to check the words after they have been stemmed. For instance "today" should stems to "todai" in the Porter Stemmer.

For test 2, one thing that is never specified is how we are supposed to pick segments to merge. In your test, you merge the segments in the logical fashion however we are unsure if everyone will merge that way. Without explicit specification, a team may fail your test while still merging correctly.

And you guys also need to make sure you clean up after each of your test cases.

**Comment by omidkv** [comment]
For my comment on test2 about merges, Zuozhi just specified the merging pattern so I think your test2 is alright based on what has been posted to slack.

**Comment by Andresheep** [comment]
> Response from team 14:
> 
> Your test looks alright. For test 1, I think one issue that we saw was that if the document numbering system is different for everyone's personal representation this test may cause issues for them. It makes sense that everyone increases document numbers as they go along but it could be a problem. I think that we also need to check the words after they have been stemmed. For instance "today" should stems to "todai" in the Porter Stemmer.
> 
> For test 2, one thing that is never specified is how we are supposed to pick segments to merge. In your test, you merge the segments in the logical fashion however we are unsure if everyone will merge that way. Without explicit specification, a team may fail your test while still merging correctly.
> 
> And you guys also need to make sure you clean up after each of your test cases.

If u mean deleting all the files by cleaning up, I think we use @after in junit to delete them after each test case. Maybe u mean something else to clean ?

File: src/test/java/edu/uci/ics/cs221/index/inverted/Team10MergeSegmentTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        Document doc3 = new Document("a dog is there");
+        invertedmanager.addDocument(doc0);
+        invertedmanager.addDocument(doc1);
+        invertedmanager.flush(); // segment 0
+        invertedmanager.addDocument(doc2);
+        invertedmanager.addDocument(doc3);
+        invertedmanager.flush(); // segment 1
+
+        Map<String,List<Integer>> expectedlist = new HashMap<>();
+        expectedlist.put("what",Arrays.asList(0,1));
```
[the remark this answers is not in the document]
Comment (Andresheep) [reply src/test/java/edu/uci/ics/cs221/index/inverted/Team10MergeSegmentTest.java]: Thx! We will change it!

**Comment by omidkv** [comment]
> > Response from team 14:
> > Your test looks alright. For test 1, I think one issue that we saw was that if the document numbering system is different for everyone's personal representation this test may cause issues for them. It makes sense that everyone increases document numbers as they go along but it could be a problem. I think that we also need to check the words after they have been stemmed. For instance "today" should stems to "todai" in the Porter Stemmer.
> > For test 2, one thing that is never specified is how we are supposed to pick segments to merge. In your test, you merge the segments in the logical fashion however we are unsure if everyone will merge that way. Without explicit specification, a team may fail your test while still merging correctly.
> > And you guys also need to make sure you clean up after each of your test cases.
> 
> If u mean deleting all the files by cleaning up, I think we use @after in junit to delete them after each test case. Maybe u mean something else to clean ?
> Also thx for noticing the todai issue. We will change that.

My bad, I didn't see it.

**Pull request merged**